app/services/listing_index_service/search/sphinx_adapter.rb

```diff
@@ -19,6 +19,12 @@ def search(community_id:, search:, includes:)
         Result::Error.new(ArgumentError.new("Both DB query and search engine would be needed to fulfill the search"))
       end
 
+      # rename listing_shape_ids to singular so that Sphinx understands it
+      if search[:listing_shape_ids] && search[:listing_shape_id].nil?
+        search[:listing_shape_id] = search[:listing_shape_ids]
+        search.delete(:listing_shape_ids)
+      end
+
```

Comment:
More elegant way to rename keys:

```ruby
search = HashUtils.rename_keys({:listing_shape_ids => :listing_shape_id}, search)
```

(note about style: I usually prefer the new Hash syntax `{:listing_shape_ids => :listing_shape_id}`, put in some cases where the Hash describes a mapping **from** some value **to** another, then I prefer using the `=>` format because the arrow strongly points the direction of the mapping)

You should also move this before line 18 (`if ... needs_search?`). The `needs_search?` methods uses `:listing_shape_id`